curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -257,8 +258,7 @@ public void stateChanged(CuratorFramework client, ConnectionState newState)
 
             client.start();
 
-            executorService = Executors.newFixedThreadPool(2, threadFactory);  // 1 for listeners, 1 for background ops
-
+            executorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
```

Comment:
The comment was wrong. This must have changed at some point. I noticed only 1 was being used and corrected it.